tools/eslint-plugin/dist/rules/public-extension-exports.js

```diff
@@ -0,0 +1,236 @@
+        return;
+      }
+
+      fs.writeFileSync(apiFilePath, declaration.symbol.escapedName + trailer, { flag: "a" });
+    }
+
+    function getFileName(parent) {
+      let currentParent = parent;
+      while (currentParent) {
+        if (currentParent.fileName !== undefined)
+          return currentParent.fileName;
+        currentParent = currentParent.parent;
+      }
+      return undefined;
+    }
+
+    function isLocalFile(declaration) {
+      if (declaration) {
+        const fileName = getFileName(declaration.parent);
+        if (fileName && typeof fileName === "string" && !fileName.includes("node_modules"))
+          return true;
+      }
+      return false;
+    }
+
+    function getParentSymbolName(declaration) {
+      if (declaration.parent && declaration.parent.symbol && !declaration.parent.symbol.escapedName.startsWith('"'))
+        return declaration.parent.symbol.escapedName;
+      return undefined;
+    }
+
+    function checkJsDoc(declaration, node) {
+      // Only check local elements, not consumed ones
+      if (!declaration || !declaration.jsDoc || !isLocalFile(declaration))
+        return undefined;
+
+      for (const jsDoc of declaration.jsDoc)
+        if (jsDoc.tags) {
+          let jsDocExtensionTag = jsDoc.tags.find(tag => tag.tagName.escapedText === extensionApiTag);
+          // Has extension API tag
+          if (jsDocExtensionTag) {
+            addToApiList(declaration, jsDocExtensionTag);
+            // Does not have any of the required release tags
+            if (!jsDoc.tags.some(tag => releaseTags.includes(tag.tagName.escapedText))) {
+              let name;
+              if (declaration.kind === ts.SyntaxKind.Constructor)
+                name = declaration.parent.symbol.escapedName;
+              else {
+                name = declaration.symbol.escapedName;
+                const parentSymbol = getParentSymbolName(declaration);
+                if (parentSymbol)
+                  name = `${parentSymbol}.${name}`;
+              }
+
+              context.report({
+                node,
+                messageId: "forbidden",
+                data: {
+                  kind: syntaxKindFriendlyNames.hasOwnProperty(declaration.kind) ? syntaxKindFriendlyNames[declaration.kind] : "unknown object type " + declaration.kind,
+                  name,
+                  releaseTags: releaseTags,
+                }
+              });
+            }
+          }
+        }
+    }
+
+    function checkWithParent(declaration, node) {
+      if (!declaration)
+        return;
+      checkJsDoc(declaration, node);
+      if (declaration.parent && [
+        ts.SyntaxKind.ClassDeclaration,
+        ts.SyntaxKind.EnumDeclaration,
+        ts.SyntaxKind.InterfaceDeclaration,
+        ts.SyntaxKind.ModuleDeclaration,
+      ].includes(declaration.parent.kind))
+        checkJsDoc(declaration.parent, node);
```

Comment:
your ts and es nodes get out of sync here, is this intended?

Reply:
@paulius-valiunas can you elaborate?

Reply:
the ts node (`declaration.parent`) doesn't represent the same node as `node` from eslint parser, but I now realize it's only used for checking jsdoc comments on the parent while errors are still reported on the original node.